File: 04_sorting/04_07_insertion_sort_recursive/insertion_sort_recursive.js
const insertionSortRecursive = (arr = [], n) => {
    if (n >= arr.length) {
        return;
    }
    for (let j_index = n; j_index > 0; j_index--) {
        if (arr[j_index] < arr[j_index - 1]) {
            [arr[j_index], arr[j_index - 1]] = [arr[j_index - 1], arr[j_index]];
        }
    }

    insertionSortRecursive(arr, n+1);
}

const array = [13, 46, 24, 52, 20, 9]

console.log("Before:-");
console.log(array);
insertionSortRecursive(array, 0)
console.log("After:-");
console.log(array);

File: 04_sorting/04_07_insertion_sort_recursive/insertion_sort_recursive.py
def insertionSortRecursive(arr: list, n: int):
    if len(arr) <= n:
        return
    for j in range(n, 0, -1):
        if arr[j] < arr[j-1]:
            arr[j], arr[j-1] = arr[j-1], arr[j]
            
    insertionSortRecursive(arr, n + 1)
                
                
elements = [13, 46, 24, 52, 20, 9]
print("Before:-")
print(elements)
insertionSortRecursive(elements, 0)
print("After:-")
print(elements)